### PR TITLE
More socket events

### LINKED ISSUE #1108: Update Bid History for lot currently being viewed by user
Update [Bid History](https://github.com/artsy/eigen/issues/1082) for lot currently being viewed by user.

### PR DESCRIPTION
**Pull request by ashfurrow**
This adds events to the event history under the current lot, formatting for currencies throughout the UI, makes the LiveAuctionBidButton now handle its own enabled state based on its button state observable, and simplifies a few things in various view models. 

Fixes #1108.

### CODE REVIEW

File: Artsy/View_Controllers/Live_Auctions/LiveActionStateReconciler.swift [hunk 1]
```diff
@@ -87,7 +88,7 @@ extension PublicFunctions: LiveAuctionStateReconcilerType {
     }
 
     var currentLotSignal: Observable<LiveAuctionLotViewModelType?> {
-        return _currentLotSignal.map { $0.flatMap { $0 } }
+        return _currentLotSignal.map { $0 as LiveAuctionLotViewModelType? }
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveActionStateReconciler.swift]: yeah, doesn't need a comment now 👍

File: Artsy/Models/API_Models/Live/LiveEvent.h [hunk 2]
```diff
@@ -25,7 +25,7 @@ typedef NS_ENUM(NSInteger, LiveEventType) {
 
 // We do this to expose these to its children
 
-@property (nonatomic, assign, readonly) NSInteger amountCents;
+@property (nonatomic, assign, readonly) UInt64 amountCents;
```
Comment (orta) [inline Artsy/Models/API_Models/Live/LiveEvent.h]: great, must have missed that

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionBidHistoryViewController.swift [hunk 3]
```diff
@@ -23,9 +25,62 @@ class LiveAuctionHistoryCell: UITableViewCell {
...
+    
+    required init?(coder aDecoder: NSCoder) {
+        return nil
+    }
+
+    deinit {
+        newEventsBeganSubscription.0.unsubscribe(newEventsBeganSubscription.1)
+        newEventsEndedSubscription.0.unsubscribe(newEventsEndedSubscription.1)
+        newEventsSubscription.0.unsubscribe(newEventsSubscription.1)
+    }
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionBidHistoryViewController.swift]: these could be wrapped up as a struct, that can then hold both parts of this tuple via a generic thingy - then unsubscribe could be done on that struct

**Comment by orta** [comment]
Great great 👍

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionBidHistoryViewController.swift (hunk 3 above)
Comment (ashfurrow) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionBidHistoryViewController.swift]: Tracking a conversation here with ideas: https://github.com/JensRavens/Interstellar/issues/40 I like the idea of a struct, we could move it to be a part of eigen if it doesn't fall into Interstellar 👍

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
        
            2 Warnings
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Needs testing on a Phone if change is non-trivial</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
    
    
  </tbody>
</table>
  

  



Time | Class | Function |
| --- | ----- | ----- |259.4ms | [AuctionViewController.swift](/artsy/eigen/tree/socket-bid-events/Artsy/View_Controllers/Auction/AuctionViewController.swift#L176) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
185.3ms | [WebSocket.swift](/artsy/eigen/tree/socket-bid-events/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
60.1ms | [RefinementOptionsViewController+Private.swift](/artsy/eigen/tree/socket-bid-events/Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift#L63) | func stackView() -> ORStackView




<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

**Pull request merged**